File: Daily/MARCH/24_03_3169/3169.js
var countDays = function(days, meetings) {
    if (meetings.length === 0) {
        return days;
    }

    // meetings.sort((a -b) => a[0] - b[0]);git 
    meetings.sort((a, b) => a[0] - b[0]);

    const mergedIntervals = [];
    let currentStart = meetings[0][0];
    let currentEnd = meetings[0][1];

    for  (let i = 1; i < meetings.length; i++) {
        const nextStart = meetings[i][0];
        const nextEnd = meetings[i][1];


    if (nextStart <= currentEnd) {
        currentEnd = Math.max(currentEnd , nextEnd);
    } else {
        mergedIntervals.push([currentStart, currentEnd]);
        currentStart = nextStart;
        currentEnd = nextEnd;
    }
    }
    mergedIntervals.push([currentStart, currentEnd]);

    let freeDays = days;
    for (const interval of mergedIntervals) {
        freeDays -= (interrval[1] - interval[0] + 1);
        }
        return Math.max(0, freeDays);
};

File: Daily/MARCH/24_03_3169/3169.py
class Solution:
    def countDays(self, days: int, meetings: list[list[int]]) -> int:
        occupied_intervals = []
        for start, end in meetings:
            occupied_intervals.append((start, end))

        if not occupied_intervals:
            return days

        occupied_intervals.sort(key=lambda x: x[0])

        merged_intervals = []
        if occupied_intervals:
            current_start, current_end = occupied_intervals[0]
            for i in range(1, len(occupied_intervals)):
                next_start, next_end = occupied_intervals[i]
                if next_start <= current_end:
                    current_end = max(current_end, next_end)
                else:
                    merged_intervals.append((current_start, current_end))
                    current_start, current_end = next_start, next_end
            merged_intervals.append((current_start, current_end))

        free_days = days

        for start, end in merged_intervals:
            free_days -= (end - start + 1)

        return max(0, free_days)